kolibri/core/exams/models.py

```diff
@@ -139,32 +129,114 @@ class Exam(AbstractFacilityDataModel):
     #   by their user's UUID.
     learners_see_fixed_order = models.BooleanField(default=False)
 
-    # Is this exam currently active and visible to students to whom it is assigned?
-    active = models.BooleanField(default=False)
-
     # Exams are scoped to a particular class (usually) as they are associated with a Coach
     # who creates them in the context of their class, this stores that relationship but does
     # not assign exam itself to the class - for that see the ExamAssignment model.
     collection = models.ForeignKey(
         Collection,
-        related_name="exams",
+        related_name="%(class)ss",
         blank=False,
         null=False,
         on_delete=models.CASCADE,
     )
     creator = models.ForeignKey(
         FacilityUser,
-        related_name="exams",
+        related_name="%(class)ss",
         blank=False,
         null=True,
         on_delete=models.CASCADE,
     )
 
-    # To be set True when the quiz is first set to active=True
-    date_activated = models.DateTimeField(default=None, null=True, blank=True)
+    """
+    As we evolve this model in ways that migrations can't handle, certain fields may
+    become deprecated, and other fields may need to be interpreted differently. This
+    may happen when multiple versions of the model need to coexist in the same database.
+
+    The 'data_model_version' field is used to keep track of the version of the model.
+
+    Certain fields that are only relevant for older model versions get prefixed
+    with their version numbers.
+    """
+    data_model_version = models.SmallIntegerField(default=3)
 
     date_created = models.DateTimeField(auto_now_add=True, null=True)
 
+    def __str__(self):
+        return self.title
+
+    def get_questions(self):
```

Comment:
Are `get_*` methods automagically made into getters for the field of the same name? So `self.questions` is the same as calling `self.get_questions()`?